### ISSUE TITLE
04 What exactly is nullptr

### ISSUE THREAD

**Issue by zwkcoding** [zwkcoding/100Days-Of-Cplusplus-bash#4]

**Comment by zwkcoding** [comment]
ref:  https://stackoverflow.com/questions/1282295/what-exactly-is-nullptr

use nullptr:
-  an integral null pointer constant can be converted to `std::nullptr_t`. The opposite direction is not allowed. This allows overloading a function for both pointers and integers, and passing nullptr to select the pointer version. Passing NULL or 0 would confusingly select the int version.

Before C++ 11, If you wanted to call it with a null pointer, and you used NULL (i.e. the value 0), then you would call the one overloaded for int:

```
void func(int n); 
void func(char *s);

func( NULL ); // guess which function gets called?
```
  f(NULL) Equals to f(0). Calls f(int).
```
void f(int);
void f(char*);

void g() 
{
  f(0); // Calls f(int).
  f(NULL); // Equals to f(0). Calls f(int).
}
```
This is probably not what you wanted. C++11 solves this with nullptr; Now you can write the following:

```
void g()
{
  f(nullptr); //calls f(char*)
}
```

**In other words, wherever you were writing NULL before, you should use nullptr instead**
